### PR TITLE
Mobile app: Redirect to clan when tapping on the invite link

### PR DESCRIPTION
**Pull request by loi-huynhphuc**
Mobile app: Redirect to clan when tapping on the invite link https://github.com/nccasia/mezon-fe/issues/3891

### CODE REVIEW

File: apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx [hunk 1]
```diff
@@ -417,22 +423,26 @@ const MessageItem = React.memo(
...
-									translate={t}
-									onMention={onMention}
-									onChannelMention={onChannelMention}
-									isNumberOfLine={isNumberOfLine}
-									isMessageReply={false}
-									mode={mode}
-									directMessageId={channelId}
-									isOnlyContainEmoji={isOnlyContainEmoji}
-								/>
+								{isInviteLink ? (
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx:427]: !!isInviteLink boolean

File: apps/mobile/src/app/screens/home/homedrawer/components/JoinClanModal/index.tsx [hunk 2]
```diff
@@ -5,16 +5,16 @@ import {
...
 import { Block, size } from '@mezon/mobile-ui';
 import { channelsActions, clansActions, getStoreAsync } from '@mezon/store-mobile';
 import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Keyboard, KeyboardAvoidingView, Platform, Text, TouchableOpacity, TouchableWithoutFeedback } from 'react-native';
+import { Keyboard, Text, TouchableOpacity, TouchableWithoutFeedback } from 'react-native';
 import { ErrorInput } from '../../../../../components/ErrorInput';
 import { MezonInput, MezonModal } from '../../../../../temp-ui';
-import { validLinkInviteRegex } from '../../../../../utils/helpers';
+import { inviteLinkRegex, validLinkInviteRegex } from '../../../../../utils/helpers';
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/components/JoinClanModal/index.tsx]: move sang helpers của lib `inviteLinkRegex, validLinkInviteRegex`

File: apps/mobile/src/app/screens/home/homedrawer/components/RenderMessageInvite/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { style } from './RenderMessageInvite.styles';
+
+interface IRenderMessageInviteProps {
+	content: string;
+}
+
+function RenderMessageInvite({ content }: IRenderMessageInviteProps) {
+	const { themeValue } = useTheme();
+	const styles = style(themeValue);
+	const parts = content?.split(urlRegex);
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/components/RenderMessageInvite/index.tsx]: use useMemo to split

File: apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx (hunk 1 above)
Comment (loi-huynhphuc) [reply apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx:427]: chỗ này mình save thì eslint nó tự covert thì isInviteLink  luôn đó anh

**Review by phuocnguyenncc — APPROVED** [review]

**Pull request merged**